doc/topics/core-structure.md

```diff
@@ -0,0 +1,202 @@
+Node.js also has some internal files, for utilities not intended to be exposed.
+Those can be found in the `lib/internal/` folder, and files within are not
+accessible to JavaScript code outside of the node core.
+
+Additionally, Node.js's own JavaScript entry point can be found at
+`lib/internal/bootstrap_node.js`. This file determines how to execute the user
+code, and also helps set up the `process` global object, importing lots from
+`lib/internal/process/`.
+
+As you inspect these files, you will occasionally see code that calls
+`process.binding()` to import additional code or data. This leads us into the
+next segment...
+
+
+### C++ Source (`src/`)
+
+Node.js is built ontop of a C++ base for both bootstrapping and performance
+reasons. This also helps for interoperability with Node.js's underlying
+JavaScript engine, V8, which is also written in C++.
+
+Some of the C++ source files export JavaScript code defined in C++ via the V8
+API with the help of internal node helpers. These files are the "glue" between
+the JavaScript & C++ layers, and often times act as a proxy between JS and
+Libuv, the I/O library written in C which node builds atop of. The files which
+are exposed to `process.binding()` all do so via a
+`NODE_MODULE_CONTEXT_AWARE_BUILTIN` Macro, and are able to be found by doing a
+search of the folder.
+
+The contents of the `src/` folder can generalized into 3 parts:
+- `node_<subsystem>` files (Usually helpers exposed to JS)
+- `<subsystem>_wrap` files (Libuv wrappers for Proxying to/from JS APIs)
+- Other bootstrapping / utility `.cc`/`.h` files, particularly `node.cc` and `env-inl.h`.
+
+The following traces Node.js through startup, showing how it interacts with
+C++ & JS:
+
+Step | Description | Source Location
+---|---|---
+1 | startup: `main --> node::Start --> node::StartNodeInstance` | ./src/node_main.cc::51, ./src/node.cc::4345, ./src/node.cc::4383
+2 | create JSVM (`v8::isolate`) | ./src/node.cc::4372, ./src/node.cc::4261
+3 | create event loop (`uv_loop_t`) | ./src/node.cc::4377
+4 | create native environment (`node::Environment`) containing JSVM and event loop | ./src/node.cc::4276, ./src/node.cc::4181
+5 | create JS `process` object bound to native environment | ./src/node.cc::4237-4244
+6 | initialize JS context and `process` object through JS script | ./src/node.cc::4289, ./src/node.cc::3330, ./lib/internal/bootstrap_node.js
+7 | invoke user script (or repl, --eval, etc.) | ./lib/internal/bootstrap_node.js::66
+
+
+### Tests & Benchmarks (`test/` & `benchmark/`)
+
+Node.js's extensive test suite can be found within `test/`. Most new & existing
+tests can be found within `test/parallel/`, which is the directory we tell the
+test runner to parallelize over so that the test suite can run faster.
+
+Most times if you modify code within node, tests for that behavior will also be
+needed for the change to be accepted.
+
+The tests can be run via `make -j4 test` on Unix, or `vcbuild test` on Windows.
```

Comment:
`-j` is not the number of processors, but jobs. You can actually oversaturate since the OS will handle it anyhow. Only once I got an insufficient memory error and once insufficient disk space. `-j8` seems to work best in terms of speed for me on any laptop and is the most widespread one as far as I can tell. I don't mind it too much tho.